Scripts/DCS-BIOS/lib/modules/aircraft_modules/F-4E.lua

```diff
@@ -1517,7 +1517,59 @@ F_4E:defineFullRangeFloat("PLT_SHAKE_GLARESHIELD_TWIST", 3033, PILOT_SHAKE_EFFEC
+F_4E:defineToggleSwitch("WSO_TGT_DESIGNATOR_WRCS_OUT", TARGET_DESIGNATOR_DEVICE_ID, 3012, 2757, WSO_TARGET_DESIGNATOR_SET_CONTROL, "Enable/Disable WRCS integration")
+F_4E:defineToggleSwitch("WSO_TGT_DESIGNATOR_GO_MALF", TARGET_DESIGNATOR_DEVICE_ID, 3013, 2758, WSO_TARGET_DESIGNATOR_SET_CONTROL, "(no function)")
+F_4E:defineToggleSwitch("WSO_TGT_DESIGNATOR_INS_OUT", TARGET_DESIGNATOR_DEVICE_ID, 3014, 2759, WSO_TARGET_DESIGNATOR_SET_CONTROL, "Enable/Disable INS integration")
+
+F_4E:defineIndicatorLight("WSO_TGT_DESIGNATOR_LIGHT_STOW", 2715, WSO_TARGET_DESIGNATOR_SET_CONTROL, "Target Designator Stow button lamp")
+F_4E:defineIndicatorLight("WSO_TGT_DESIGNATOR_LIGHT_LASER_READY", 2716, WSO_TARGET_DESIGNATOR_SET_CONTROL, "Target Designator Laser Ready button lamp")
+F_4E:defineIndicatorLight("WSO_TGT_DESIGNATOR_LIGHT_POWER_ON", 2717, WSO_TARGET_DESIGNATOR_SET_CONTROL, "Target Designator Power On button lamp")
+F_4E:defineIndicatorLight("WSO_TGT_DESIGNATOR_LIGHT_WRCS_OUT", 2718, WSO_TARGET_DESIGNATOR_SET_CONTROL, "Target Designator WRCS button lamp")
+F_4E:defineIndicatorLight("WSO_TGT_DESIGNATOR_LIGHT_GO", 2719, WSO_TARGET_DESIGNATOR_SET_CONTROL, "Target Designator GO button lamp")
+F_4E:defineIndicatorLight("WSO_TGT_DESIGNATOR_LIGHT_MALF", 2720, WSO_TARGET_DESIGNATOR_SET_CONTROL, "Target Designator MALF button lamp")
+F_4E:defineIndicatorLight("WSO_TGT_DESIGNATOR_LIGHT_OVHT", 2721, WSO_TARGET_DESIGNATOR_SET_CONTROL, "Target Designator Overheat button lamp")
+F_4E:defineIndicatorLight("WSO_TGT_DESIGNATOR_LIGHT_INS_OUT", 2722, WSO_TARGET_DESIGNATOR_SET_CONTROL, "Target Designator INS Out button lamp")
+
+-- Laser Coder Control
+local WSO_LASER_CODE_PANEL = "WSO Laser Code Panel"
+
+F_4E:definePushButton("WSO_LASER_CODE_ENTER", TARGET_DESIGNATOR_DEVICE_ID, 3015, 2504, WSO_LASER_CODE_PANEL, "Enter Laser Code")
+F_4E:definePushButton("WSO_LASER_CODE_SET_THOUSANDS", TARGET_DESIGNATOR_DEVICE_ID, 3019, 2500, WSO_LASER_CODE_PANEL, "Next Laser Code (thousands)")
+F_4E:definePushButton("WSO_LASER_CODE_SET_HUNDREDS", TARGET_DESIGNATOR_DEVICE_ID, 3018, 2501, WSO_LASER_CODE_PANEL, "Next Laser Code (hundreds)")
+F_4E:definePushButton("WSO_LASER_CODE_SET_TENS", TARGET_DESIGNATOR_DEVICE_ID, 3017, 2502, WSO_LASER_CODE_PANEL, "Next Laser Code (tens)")
+F_4E:definePushButton("WSO_LASER_CODE_SET_ONES", TARGET_DESIGNATOR_DEVICE_ID, 3016, 2503, WSO_LASER_CODE_PANEL, "Next Laser Code (ones)")
+F_4E:defineIndicatorLight("WSO_LIGHT_LASER_NO_GO", 2697, WSO_LASER_CODE_PANEL, "Laser No Go button lamp")
+F_4E:defineIntegerFromArg("WSO_LASER_CODE_ONES", 2020, 9, WSO_LASER_CODE_PANEL, "Laser Code (ones)")
+F_4E:defineIntegerFromArg("WSO_LASER_CODE_TENS", 2021, 9, WSO_LASER_CODE_PANEL, "Laser Code (tens)")
+F_4E:defineIntegerFromArg("WSO_LASER_CODE_HUNDREDS", 2022, 9, WSO_LASER_CODE_PANEL, "Laser Code (hundreds)")
+F_4E:defineIntegerFromArg("WSO_LASER_CODE_THOUSANDS", 2023, 9, WSO_LASER_CODE_PANEL, "Laser Code (thousands)")
```

Comment:
Just tested these, and these are off by one at values > 5. The drum values have this weird behavior where both 0 and 1 in float range map to a drum value of 0. But if you set the max value to 10, then you get weird behavior where bios outputs 10 instead of 0. The `drum_value` function handles this properly.


```suggestion
F_4E:defineIntegerFromGetter("WSO_LASER_CODE_ONES", function(dev0)
	return drum_value(dev0, 2020)
end, 9, WSO_LASER_CODE_PANEL, "Laser Code (ones)")
F_4E:defineIntegerFromGetter("WSO_LASER_CODE_TENS", function(dev0)
	return drum_value(dev0, 2021)
end, 9, WSO_LASER_CODE_PANEL, "Laser Code (tens)")
F_4E:defineIntegerFromGetter("WSO_LASER_CODE_HUNDREDS", function(dev0)
	return drum_value(dev0, 2022)
end, 9, WSO_LASER_CODE_PANEL, "Laser Code (hundreds)")
F_4E:defineIntegerFromGetter("WSO_LASER_CODE_THOUSANDS", function(dev0)
	return drum_value(dev0, 2023)
end, 9, WSO_LASER_CODE_PANEL, "Laser Code (thousands)")
```

Reply:
Strange, I didn't see this behavior, but I tested with my external application, which subscribe for particular control e.g. `WSO_LASER_CODE_ONES` and notify me when value changed.